migrations/migration.py

```diff
@@ -0,0 +1,139 @@
+import os
+from typing import List
+from datetime import datetime
+
+from isacc_messaging.audit import audit_entry
+from isacc_messaging.models.fhir import (
+    HAPI_request,
+    first_in_bundle,
+    next_in_bundle,
+    resolve_reference,
+)
+from migrations.migration_resource import MigrationManager
+
+class Migration:
+    def __init__(self):
+        self.migrations_dir = os.path.join(os.path.dirname(__file__), "versions")
+        self.latest_applied_migration = self.get_applied_migration_from_fhir()
+
+    def get_migration_files(self) -> List[str]:
+        """Retrieve the list of migration files."""
+        migration_files = sorted(os.listdir(self.migrations_dir))
+        return migration_files
+
+    def get_latest_migration(self) -> int:
+        """Find the latest migration number."""
+        migration_files = self.get_migration_files()
+        if migration_files:
+            latest_migration = int(migration_files[-1].split("_")[0])
+            return latest_migration
+        else:
+            return 0
+
+    def get_applied_migration_from_fhir(self) -> int:
+        """Retrieve the latest applied migration number from FHIR."""
+        # Logic to retrieve the latest applied migration number from FHIR
+        basic = HAPI_request('GET', 'Basic', params={
+            "identifier": f"http://isacc.app/twilio-message-sid|{id}"
+        })
+        basic = first_in_bundle(basic)
+
+        if basic is None:
+            self.create_new_migration_manager()
+
+            return 1
+
+        manager = MigrationManager(basic)
+        latest_applied_migration = manager.get_migration()
+
+        return latest_applied_migration
+
+    def update_applied_migration_in_fhir(self, latest_applied_migration: int):
```

Comment:
i probably confused the issue, using simple integers when discussing the examples for how to order.  the problem with using integers is there's no flexibility in reordering - say two dev branches grow apart and then come back together with batches of upgrade steps in each.  you wouldn't want to hand adjust a whole batch of revision and back_revision values.

a random hash string as the revision number solves this.  the order is just defined by the respective `down_revision` values in the scripts.  to reorder a fork, only a single down_revision edit is needed.

Reply:
That explains my questions, thank you! I was rather confused, sorry.